### DISCUSSION TITLE
Rich for diffs in the CLI?

### DISCUSSION THREAD

**Discussion by DiegoMairena** [Textualize/rich-cli#93] [Ideas]
Hey folks, 

Thanks for all the work you've put into Rich :) 

Just wanted to ask if it would be possible to have rich run diffs? 

Something like:

| csv 1     | csv 2 | diff    |
| :---      |    :--|     ---:|
| foo       |foo   |         |
| bar       |       | not in csv 2 |
| blah      |blah   |         |

Cheers

**Comment by harkabeeparolus** [comment]
I'm just a fan of Rich, like many others. However, I do work with csv files. In my opinion, _rich-cli_ should not be concerned with processing and analyzing csv files, only with displaying them.

If you're looking for was to process your csv files, there are many other tools.

* [qsv](https://github.com/jqnatividad/qsv) has a `diff` command.
* The Unix utility [comm](https://pubs.opengroup.org/onlinepubs/9699919799/utilities/comm.html) can display a diff just like you're asking for. Note that the input files _must_ be sorted.
* For Python CSV processing on the command line, I recommend [csvkit](https://github.com/wireservice/csvkit). It doesn't do diffs like you're asking for, but it's great for many other things.
  * With csvkit, you could use `csvcut` to extract and `csvsort` to sort the key columns, in order to use the `comm` utility mentioned above. That's how I would solve this task.

**Reply by DiegoMairena** [reply]
Hey, thanks for your reply. I actually did not know about some of those tools you mentioned and it does look like comm might be what I'm after. I'll definitely give it a good go.

Cheers
